server/api/cli/cli.controller.js

```diff
@@ -39,17 +39,22 @@ exports.run = function (req, res) {
   dirName += dateTime.getFullYear()
   const execFile = req.body.fileName.replace('.c', '')
   const cmd = `"${dirName}/${execFile}"`
-  exec(cmd, { timeout: 10000 }, // Process will time out if running for > 10 seconds.
+  var cp = exec(cmd, { timeout: 10000 }, // Process will time out if running for > 10 seconds.
     (error, stdout, stderr) => {
       if (error) {
-        return res.send(200, error)
+        return res.send(200, {"info":error, "process":cp.pid});
```

Comment:
I just wrote a little program to test.

exec.js
```javascript
const exec = require('child_process').exec

var cp = exec('loop.bat', { timeout: 10000 }, // Process will time out if running for > 10 seconds.
    (error, stdout, stderr) => {
      if (error) {
      console.log({"info":error, "process":cp.pid});
      }
      console.log({"info":stdout, "process":cp.pid});
    })
```

loop.bat
```
@echo off
:LOOP
goto :LOOP
```

This does not print the error message until after the timeout. Therefore, the issue is probably caused by Express returning a response before the timeout.